src/ucp/core/ucp_am.c

```diff
@@ -746,14 +748,21 @@ UCS_PROFILE_FUNC(ucs_status_t, ucp_proto_progress_am_rndv_rts, (self),
 {
     ucp_request_t *sreq = ucs_container_of(self, ucp_request_t, send.uct);
     size_t max_rts_size;
+    ucs_status_t status;
+
+    ucp_request_send_track(sreq);
```

Comment:
> could just start tracking if ucp_do_am_single returns ok

no, it will be failed, since UCP will try to untrack request that is not being tracking

Reply:
maybe disable errhandling tests for self? It may simplify several places in the code like here

Reply:
> maybe disable errhandling tests for self? It may simplify several places in the code like here

we will disable tests, but we will need to add a check for AM transport used in RTR/ATS handler (to not untrack request for SELF transport)

Reply:
can you please elaborate?

Reply:
> can you please elaborate?

Here we are adding a UCP request to be tracked by a UCP EP after sending RTS to a peer.
Since we want to stop tracking of the UCP request on receiving either RTR (send request will be tracked by UCT transport that will do PUT) or ATS (operation completed successfully), RTR/ATS handler calls `ucp_request_send_untrack()`.
So, since we don't add the UCP request for being tracked before sending RTS*, we have to check whether the AM transport is SELF or not in RTR/ATS handler to make decision if we need to untrack the request.

*we add a UCP request to be tracked by a UCP EP before doing a UCT send operation to handle the possible case when RTR/ATS packet is received inside UCT AM Bcopy function due to send-recv pattern in SELF transport.